test/unit/api/roles.spec.ts

```diff
@@ -0,0 +1,330 @@
+/* eslint-disable no-unused-expressions */
+
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import 'chai/register-should';
+import jwt from 'jsonwebtoken';
+import { NotificationCategories } from 'types';
+import { NotificationSubscription } from 'models';
+import app, { resetDatabase } from '../../../server-test';
+import { JWT_SECRET } from '../../../server/config';
+import * as modelUtils from '../../util/modelUtils';
+
+import { Errors as createErrors } from '../../../server/routes/createRole';
+import { Errors as upgradeErrors } from '../../../server/routes/upgradeMember';
+import { Errors as deleteErrors } from '../../../server/routes/deleteRole';
+
+
+chai.use(chaiHttp);
+const { expect } = chai;
+
+describe('Roles Test', () => {
+  let jwtToken;
+  let loggedInAddr;
+  let loggedInAddrId;
+  let adminUserId;
+  const chain = 'ethereum';
+  const community = 'staking';
+
+  before('reset database', async () => {
+    await resetDatabase();
+    const res = await modelUtils.createAndVerifyAddress({ chain });
+    loggedInAddr = res.address;
+    loggedInAddrId = res.address_id;
+    jwtToken = jwt.sign({ id: res.user_id, email: res.email }, JWT_SECRET);
+    adminUserId = res.user_id;
+    const isAdmin = await modelUtils.assignRole({
+      address_id: res.address_id,
+      chainOrCommObj: { offchain_community_id: community },
+      role: 'admin',
+    });
+  });
+
+  describe('/createRole route tests', () => {
+    it('should pass on joining public community', async () => {
+      const user = await modelUtils.createAndVerifyAddress({ chain });
+      const res = await chai.request(app)
+        .post('/api/createRole')
+        .set('Accept', 'application/json')
+        .send({
+          jwt: jwt.sign({ id: user.user_id, email: user.email }, JWT_SECRET),
+          community,
+          address_id: user.address_id,
+        });
+      expect(res.body.status).to.be.equal('Success');
+      expect(res.body.result.address_id).to.be.equal(user.address_id);
+      expect(res.body.result.offchain_community_id).to.be.equal(community);
+    });
+
+    it('should fail on joining public community a second time', async () => {
```

Comment:
"should fail to create duplicate role for a public community a user is a member of"